contractcourt/htlc_success_resolver.go

```diff
@@ -191,6 +205,17 @@ func (h *htlcSuccessResolver) Resolve() (ContractResolver, error) {
 		return nil, h.Checkpoint(h)
 	}
 
+	// The wallet returns resolutions with the zero preimage, as it doesn't
+	// have access to the preimage cache. Before constructing the input, we
+	// lookup the cache and apply it to our resolution.
+	preimage, ok := h.PreimageDB.LookupPreimage(h.payHash)
+	if !ok {
+		err := fmt.Errorf("unable to find preimage "+
+			"for hash=%v", h.payHash)
+		return nil, err
+	}
```

Comment:
You could move the lookup code into `applyPreimage` too and rename the method. Prevent duplication from above. But not critical.

Reply:
good suggestion, will carry this over to a subsequent PR that removes the preimage cache from lnwallet